scripts/validate_rst_title_capitalization.py

```diff
@@ -178,13 +181,64 @@ def correct_title_capitalization(title: str) -> str:
     # Split a title into a list using non-word character delimiters.
     word_list = re.split(r"\W", removed_https_title)
 
-    for word in word_list:
-        if word.lower() in CAP_EXCEPTIONS_DICT:
-            correct_title = re.sub(
-                rf"\b{word}\b", CAP_EXCEPTIONS_DICT[word.lower()], correct_title
-            )
-
-    return correct_title
+    tuple_exceptions_dict: dict = {}
+    for indx, word in enumerate(word_list):
```

Comment:
Nice work here, but I think this implementation is too complex to be maintained in pandas. We could have the title validation as a separate project (may be a sphinx extension). But I think it's probably better to instead forget about giving the user the right capitalization. If we don't care about telling the user what capitalization is correct, the implementation is quite simple.

The list of words with a specific capitalization can still be a list of strings. We check for each word in a title, it the lower case capitalization matches a word in the list of specific capitalizations (in lower case too). And if it does, we make sure that the original word is in the list of special cases (without changing the capitalization).

What do you think?

Reply:
Im sorry, Im not sure I understand. Can you explain a little bit more?

Reply:
You want to keep record of the case format of words that are in a list of exceptions?

Reply:
The original implementation we did was doing next:
- Get the right capitalization of a sentence
- Compare if it's the same as the one in the analyzed title
- If not, fail and report the right sentence to the user, so it's easy for them to update

There is a problem with it. Some words are valid with more than one capitalization. Your approach here is nice in that it keeps generating the right sentences, so they can be shown to the user. But it comes at the cost of being complex to implement.

What we can do, is simply check word by word in the analyzed sentence. If the word is in the list of known words, then we're happy with it. If it's not, then it needs to be capitalized depending on the position in the sentence (first word starts by capital letter, the rest all lower case). This is very easy to implement. We won't be able to tell the user what is the correct sentence anymore. But what I'm saying is that I'm ok with it. I think it's better to have a bit less useful error message, to save a significant amount of code.

Does it make sense?

Reply:
Yeah sure. So you want to keep it as it is as of now, if I understand correctly? My PR was a suggestion, and I understand that it's too complex.

Reply:
Not as is. I would change how it work, but instead of this implementation, removing the part where we compute the right capitalization, and simply validate that the words are correctly capitalized (what I explained in the previous comment).